core/templates/domain/exploration/WrittenTranslationObjectFactory.ts

```diff
@@ -90,11 +96,13 @@ export class WrittenTranslation {
   providedIn: 'root'
 })
 export class WrittenTranslationObjectFactory {
-  createNew(
-      type: WrittenTranslationDataFormat,
-      html: string
-  ): WrittenTranslation {
-    return new WrittenTranslation(type, html, false);
+  createNew(dataFormat: string): WrittenTranslation {
+    if (!DATA_FORMAT_TO_DEFAULT_VALUES.hasOwnProperty(dataFormat)) {
+      throw new Error('Invalid translation data format: ' + dataFormat);
+    }
+
+    return new WrittenTranslation(
+      dataFormat, DATA_FORMAT_TO_DEFAULT_VALUES[dataFormat], false);
   }
 
   createFromBackendDict(
```

Comment:
[Optional] Can you move these two functions to the `WrittenTranslation` class and make them static members. This object-factory/ Injectable is not really required anymore.